ix_challenge_list_posts_by_rating/app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  before_action :find_post, only: [:upvote, :downvote]
+
+  def create
+    post = Post.new(post_params)
+
+    if post.save
+      render json: post, status: :created
+    else
+      render json: { errors: post.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def upvote
+    @post.upvotes += 1
+    @post.save
```

Comment:
```suggestion
    @post.update(upvotes: @post.upvotes + 1)
```
existe alguma razão em atribuir o valor e depois salvar o objeto, em vez de simplesmente fazer numa só operação?

Reply:
Eu estou acostumado a setar os valores no objeto antes e salvar por último. Mas a sugestão de fazer em uma só linha, me parece ótima. Vou aderir.